src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -162,37 +147,8 @@ export function fetchAll(
       // but their errors will be shown in-place (e.g. of the chart).
       .catch(sendErrorTo(dataSubjects.documents$, dataSubjects.main$));
 
-    charts
-      ?.then((chart) => {
-        dataSubjects.totalHits$.next({
-          fetchStatus: FetchStatus.COMPLETE,
-          result: chart.totalHits,
-          recordRawType,
-        });
-
-        dataSubjects.charts$.next({
-          fetchStatus: FetchStatus.COMPLETE,
-          response: chart.response,
-          recordRawType,
-        });
-
-        checkHitCount(chart.totalHits);
-      })
-      .catch(sendErrorTo(dataSubjects.charts$, dataSubjects.totalHits$));
-
-    totalHits
-      ?.then((hitCount) => {
-        dataSubjects.totalHits$.next({
-          fetchStatus: FetchStatus.COMPLETE,
-          result: hitCount,
-          recordRawType,
-        });
-        checkHitCount(hitCount);
-      })
-      .catch(sendErrorTo(dataSubjects.totalHits$));
-
     // Return a promise that will resolve once all the requests have finished or failed
-    return Promise.allSettled([documents, charts, totalHits]).then(() => {
+    return documents.then(() => {
```

Comment:
Ok, we need to rewrite this so the promise is resolved when documents AND hits are returned. Why? e.g. for when an interval is set, so the next interval triggered request doesn't start if the histogram doesn't successfully submit "hits ready"

Reply:
Hmm good point and something I didn't consider before. I'll look into this more today.

Reply:
this works not as expected, thx a lot!